front-end/src/components/RecordingControls.js

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from 'react';
+
+class RecordingControls extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            mediaRecorder: null,
+            recording: false
+        }
+    }
+
+    componentDidMount() {
+        navigator.mediaDevices.getUserMedia({ audio: true })
+            .then(stream => {
+                console.log('Stream connected');
+                let mediaRecorder = new MediaRecorder(stream);
+                this.setState({ mediaRecorder });
+            })
+    }
+
+    _handleStartRecording = () => {
+        this.state.mediaRecorder.start();
+        this.setState({ recording: true });
+        console.log(this.state.mediaRecorder.state)
+    }
+
+    _handleStopRecording = () => {
+        this.setState({ recording: false });
+        let chunks = [];
+        let mediaRecorder = this.state.mediaRecorder;
+        mediaRecorder.stop();
+        console.log(this.state.mediaRecorder.state)
+        mediaRecorder.ondataavailable = event => {
+            chunks.push(event.data);
+        }
+        mediaRecorder.onstop = (event) => {
+            let blob = new Blob(chunks, { 'type' : 'audio/mp3; codecs=opus' });
+            let audioURL = window.URL.createObjectURL(blob);
+            console.log(audioURL);
+            this.setState({ audioURL });
+        }
+        
+    }
+
+    render() {
+        return (
+            <div>
+                {this.state.recording
```

Comment:
Is there a shorter/implicitly documented solution that does the same thing as this ternary operator? The only difference is the className.

Reply:
I made it a bit more clear as to what was happening with changes to the css naming. Also, because we don't want onClick functionality to the button while it's in a recording state, it may be simpler to keep the ternary with two separate buttons instead of changing data within the element itself.

Reply:
Ah, makes sense. Looks good!